torchrl/data/tensordict/tensordict.py

```diff
@@ -3812,3 +3971,56 @@ def _check_keys(
                         f"incompatible"
                     )
     return keys
+
+
+_accepted_classes = (torch.Tensor, MemmapTensor, _TensorDict)
+
+
+def _expand_to_match_shape(parent_batch_size, tensor, self_batch_dims, self_device):
+    if hasattr(tensor, "dtype"):
+        return torch.zeros(
+            *parent_batch_size,
+            *tensor.shape[self_batch_dims:],
+            dtype=tensor.dtype,
+            device=self_device,
+        )
+    else:
+        # tensordict
+        out = TensorDict(
+            {},
+            [*parent_batch_size, *tensor.shape[self_batch_dims:]],
+            device=self_device,
+        )
+        return out
+
+
+def flatten_keys(
```

Comment:
we may eventually want to have a `unflatten_key` method as well

Reply:
makes sense I can do that

Reply:
See latest commit :)